### DISCUSSION TITLE
Проблема игровых лаунчеров при работе с методом -dpi-desync-fooling=md5sig

### DISCUSSION THREAD

**Discussion by NotMephisto** [bol-van/zapret#907] [Q&A] [answered]
Есть проблемы с методом обмана md5sig которая не даёт покоя игровым лаунчерам

В Steam не синхронизируются сохранения игр, Uplay обновление лаунчера + синхронизация, а некоторые лаунчеры вообще не запускают игру из-за фуллинга, например, modrinth 

Я его использую для обхода Discord блокировки + помогает мне обойти блокировку тунельного подключения в браузере

Можно ли сделать белый список для игровых лаунчеров чтобы их не касались обходные методы?

Если пробовать другие методы обхода, то либо отваливается тунельное подключение, либо Discord 

Провайдер Lovit

Настройка сама + правило для обхода блокировки Discord
```
NFQWS_PORTS_TCP=80,443
NFQWS_PORTS_UDP=443
NFQWS_TCP_PKT_OUT=9
NFQWS_TCP_PKT_IN=3
NFQWS_UDP_PKT_OUT=9
NFQWS_UDP_PKT_IN=0
NFQWS_PORTS_TCP_KEEPALIVE=
NFQWS_PORTS_UDP_KEEPALIVE=
NFQWS_OPT="
--filter-tcp=80 --dpi-desync=fake,split2 --dpi-desync-fooling=md5sig <HOSTLIST> --new
--filter-tcp=443 --dpi-desync=fake,disorder2 --dpi-desync-fooling=md5sig <HOSTLIST> --new
--filter-udp=443 --dpi-desync=fake --dpi-desync-repeats=6 <HOSTLIST_NOAUTO>"
```

**Comment by electrifying** [comment]
Можно использовать другой метод fooling, либо вынести все обращения лаунчеров в список исключений или поменять стратегию/сделать мульти

**Reply by NotMephisto** [reply]
Как правильнее всего внести в белый список? Я нашёл где именно пакеты тормозятся и на каких доменах это происходит, например, login.live.com не получает пакеты, поэтому Minecraft не запускается. Однако со Steam и Uplay не понятно что именно им мешает обновляться/синхронизировать данные с серверами, есть некоторые домены которые тоже стопарятся (например, video.fastly.steamstatic.com, shared.steamstatic.com, clan.steamstatic.com), однако я не уверен что именно этот домен отвечает за синхронизацию сохранений

**Reply by electrifying** [reply]
> Как правильнее всего внести в белый список?

Добавить адреса в zapret-hosts-user-exclude

**Reply by chipodrillo** [reply]
> Как правильнее всего внести в белый список? Я нашёл где именно пакеты тормозятся и на каких доменах это происходит, например, login.live.com не получает пакеты, поэтому Minecraft не запускается. Однако со Steam и Uplay не понятно что именно им мешает обновляться/синхронизировать данные с серверами, есть некоторые домены которые тоже стопарятся (например, video.fastly.steamstatic.com, shared.steamstatic.com, clan.steamstatic.com), однако я не уверен что именно этот домен отвечает за синхронизацию сохранений

А через что смотрел?
У Меня тоже проблемы с облосными сохранениями в steam и Origin.., но думал это нормально в нынешнее время... Через раз синхронизируется.., а в ориджите с такой же переодичностью в игры заходит... ошибки бывают вылетают...

**Reply by NotMephisto** [reply]
> А через что смотрел?

Через бредмауэр Portmaster

> У Меня тоже проблемы с облосными сохранениями в steam и Origin.., но думал это нормально в нынешнее время... Через раз синхронизируется.., а в ориджите с такой же переодичностью в игры заходит... ошибки бывают вылетают...

Хм, странно что у вас это именно так работает
Вы такой же метод --dpi-desync-fooling используете?

**Reply by NotMephisto** [reply]
> Добавить адреса в zapret-hosts-user-exclude

Не работает, добавил login.live.com для проверки и безуспешно

**Reply by numbereleven-a** [reply]
> > Добавить адреса в zapret-hosts-user-exclude
> 
> Не работает, добавил login.live.com для проверки и безуспешно

стоит MODE_FILTER=hostlist ? Не очень понимаю как запрет оказывает воздействие на сайты которые не в списке для дурения.
можно попробовать снять дебаглог, может он что то подскажет

**Reply by NotMephisto** [reply]
> > > Добавить адреса в zapret-hosts-user-exclude
> > 
> > 
> > Не работает, добавил login.live.com для проверки и безуспешно
> 
> стоит MODE_FILTER=hostlist ? Не очень понимаю как запрет оказывает воздействие на сайты которые не в списке для дурения. можно попробовать снять дебаглог, может он что то подскажет

Он работает со всеми запросами которые к нему приходят

В моём случаи это помогает мне обманывать там, где неожидал бы я блокировки
Из минусов лишь со синхронизацией данных в лаунчерах

Я не использую обман по спискам или по ресолвленным айпи

> можно попробовать снять дебаглог, может он что то подскажет

В дебаглоге будут записаны приходящие к нему запросы?

**Reply by numbereleven-a** [reply]
MODE_FILTER что стоит?

**Reply by numbereleven-a** [reply]
ставь autohostlist и тогда exclude листы будут работать, а дурение применяться к неожиданным блокировкам

**Reply by chipodrillo** [reply]
> Хм, странно что у вас это именно так работает Вы такой же метод --dpi-desync-fooling используете?

md5sig

**Reply by NotMephisto** [reply, marked as answer]
> ставь autohostlist и тогда exclude листы будут работать, а дурение применяться к неожиданным блокировкам

~Попробал сделать так, отвалился Discord, но остальное всё заработало~

Решил проблему добавлением своего списка заблокированных доменов в `zapret-host-auto` и включением режима фильтрации `autohostlist`

Всем спасибо за помощь!

Если кому-то понадобится файл с заблокированными доменами:
[zapret-hosts-auto.txt](https://github.com/user-attachments/files/18036243/zapret-hosts-auto.txt)

**Reply by numbereleven-a** [reply]
правильней в zapret-hosts-user.txt, но раз в ауто и устраивает то ок
незачто
